core/feature.go

```diff
@@ -71,20 +67,14 @@ func (f *Features) Init(properties *config.Properties, list ...interface{}) {
 	for i, featureName := range properties.FeatureList {
 		fields[i] = reflect.StructField{
 			Name: featurePropertyName(featureName),
-			Type: reflect.TypeOf(singleFeature{}),
+			Type: reflect.PointerTo(propsType),
```

Comment:
I guess we need to use PtrTo if we're still supporting go versions before 1.18

https://go.dev/pkg/reflect/?m=old#PointerTo